esmvalcore/cmor/check.py

```diff
@@ -464,17 +464,87 @@ def _check_generic_level_dim_names(self, key, coordinate):
             else:
                 if out_name:
                     self.report_critical(
-                        f'Generic level coordinate {key} '
-                        'has wrong standard_name '
-                        'or is not set.', )
+                        f'Generic level coordinate {key} with out_name '
+                        f'{out_name} has wrong standard_name or is not set.')
                 else:
-                    self.report_critical(self._does_msg, key, 'exist')
+                    self._check_alternative_dim_names(key)
+
+    ALTERNATIVE_GENERIC_LEV_COORDS = {
+        'alevel': {
+            'CMIP5': ['alt40', 'plevs'],
+            'CMIP6': ['alt16', 'plev3'],
+            'obs4mips': ['alt16', 'plev3'],
+        },
+        'zlevel': {
+            'CMIP3': ['pressure'],
+        },
+    }
+
+    def _check_alternative_dim_names(self, key):
+        """Check for viable alternatives to generic level coordinates.
+
+        Generic level coordinates are used to calculate high-dimensional (e.g.,
+        3D or 4D) regular level coordinates (like pressure or altitude) from
+        lower-dimensional (e.g., 2D or 1D) arrays in order to save disk space.
+        A detailed explanation can be found here:
+            https://github.com/ESMValGroup/ESMValCore/issues/1029
+
+        Only the projects CMIP3, CMIP5, CMIP6 and obs4MIPs support generic
+        level coordinates. Right now, only alternative level coordinates for
+        the atmosphere ('alevel' or 'zlevel') are supported.
+
+        Note that only the "simplest" CMOR table entry per coordinate is
+        specified (e.g., only 'plev3' for the pressure level coordinate and
+        'alt16' for the altitude coordinate). These different versions (e.g.,
+        'plev3', 'plev19', 'plev39', etc.) only differ in the requested values.
+        We are mainly interested in the metadata of the coordinates (names,
+        units), which is equal for all coordinate versions. In the DEFAULT
+        strictness or lower, differing requested values only produce a warning.
```

Comment:
So this could print  "fake" warnings? I get that 90% of people don't read them, but I still think it would be a bit confusing if you get warnings because you are comparing right or wrong requested values of plev19 to plev3. There are a few plev coords, but it's not like you have thousands of options either. Maybe you could use the `len` of the depth coord to choose which plevX could be checked against to.

Reply:
Yes, there could be "fake" warnings. The problem here is that most datasets that need this fix are observational datasets, and they don't really care about `plev19`, `plev3`, etc. at all, so even checking the `len` of the coordinate would be useful. Would it be an alternative to add an argument `check_requested` to `_check_coord` and set it to `False` in this specific case?

Reply:
I guess that would the quick solution. Or print a message that requested values will not be checked.

A more elaborate option could be to try to work with the `cmor_strict` parameter in the config developer. If `True` like in CMIP projects, check the requested values. If `False` like in OBS project, skip the check. But it's quite the work, so maybe it's something that could be considered for the future.

Reply:
I will add a message and a note to the code that this might be expanded in the future